index/store/moss/reader.go

```diff
@@ -42,10 +40,8 @@ func (r *Reader) MultiGet(keys [][]byte) ([][]byte, error) {
 	return store.MultiGet(r, keys)
 }
 
-var bigOne = big.NewInt(1)
-
 func (r *Reader) PrefixIterator(k []byte) store.KVIterator {
-	kEnd := big.NewInt(0).Add(big.NewInt(0).SetBytes(k), bigOne).Bytes()
+	kEnd := incrementBytes(k)
```

Comment:
So, there is one difference between these two implementations (that I recall).  In the previous version, we can get a longer []byte than we started with, if we overflow at each step.  In the new version, we always get back a slice of bytes the same length as we we started with.

So consider trying to increment:

`0xff 0xff 0xff`

In the old version, you should get `0x01 0x00 0x00 0x00`
In the new version, you'd get `0x00 0x00 0x00`

So:
1. This my recollection, not tested.
2. I don't remember why I thought that might matter in this case, and not the other cases.
3.  Does it actually matter in any cases?